### ISSUE TITLE
Remoting Protocol Network (Tx/RX) and packet loss 

### ISSUE THREAD

**Issue by JeremyMain** [JeremyMain/GPUProfiler#137]
Adding the remoting protocol metrics for Network Tx/Rx and packet loss information.
HDX, PCoIP and Blast support.
Request for details about TGX, RDP if anyone has data sources I can add those as well.

**Comment by JeremyMain** [comment]
HDX, PCoIP and Blast protocol metrics AND overall network utilization (Tx/Rx) implemented and tested. 
For protocols that don't have data sources the general network utilization can be used.

**Comment by stitchpc** [comment]
I see a need in the market for TGX, RDP & HP Z Central for GFX protocol support (priority for TGX as they support Game Pad). Let me know what Data source you need, I can provide samples for each. There are other key network metrics that we measure separately that would be nice to have like Jitter and packet re-ordering but most important is comparing Sender vs Receiver Tx/Rx at both ends.

**Comment by JeremyMain** [comment]
If you have those metric details please share them and I can attempt to add them to the next release (the release with the network / protocol utilization. I would need to add an additional define for ZCentral.

![image](https://user-images.githubusercontent.com/19617537/123200759-564a1c80-d4ec-11eb-9ed2-6666546f9bfd.png)

**Comment by mjnowen** [comment]
Would love to see [NICE DCV](https://aws.amazon.com/hpc/dcv/) protocol supported here. Let us know in this comment thread what information/software/resources you need to help make this happen.

**Comment by JeremyMain** [comment]
@mjnowen 
I installed DCV server in a VM and poked around WMI looking for something relevant but no luck.
If you know if there is an DCV API that I can query to collect some of the metrics that the client can display (on the server-side) it would make adding Nice DCV support possible.
![image](https://user-images.githubusercontent.com/19617537/123963918-e1944800-d9ed-11eb-92c6-a9328387d4f6.png)

**Comment by JeremyMain** [comment]
The DCV documentation and API calls haven't revealed anything as well.
https://docs.aws.amazon.com/dcv/index.html

There is a "metrics" registry key but no documentation about what key values and what that enables.
Computer\HKEY_CURRENT_USER\SOFTWARE\GSettings\com\nicesoftware\dcv\metrics

**Comment by pbor** [comment]
Yes, that is something we plan to better document once we also improve the content of the metrics we collect.

In the mean time, in that registry folder you can set a registry key called `reporters` and assign the string value `['jsonlogfile']` (including the square brackets, since potentially it is a list of reporters but right now the json dump is the only one that we have.

This will generate json log files every minute in the DCV logs folder (C:\ProgramData\NICE\dcv).
There will be a log file for each process (server and agents) and the files are rotated so you can pick them up at your own pace.

**Comment by JeremyMain** [comment]
adding ```reporters``` : ``` ['jsonlogfile'] ``` to
```Computer\HKEY_CURRENT_USER\SOFTWARE\GSettings\com\nicesoftware\dcv\metrics\```
resulted in the user session agent.%USERNAME%.json files to be generated. Not quite what I was looking to parse.

With some more digging in the registry, I added the same to
```Computer\HKEY_USERS\.DEFAULT\Software\GSettings\com\nicesoftware\dcv\metrics```
which now generates:
* agent.metrics.json
* agentsession.json
* server-metrics.json

**Comment by JeremyMain** [comment]
Current status is that overall network Tx/Rx as well as remoting protocol Tx/Rx for some protocols (PCoIP, Blast, HDX) are in verification phase.
WIP screenshot, there are some other UI issues that need to still be resolved:
Img Tx is the Tx value for the remoting protocol's image transfer counter.
![image](https://user-images.githubusercontent.com/19617537/132703749-97c4c128-e1bf-433a-a365-dd892ee3fdf7.png)
